src/corelibs/U2Algorithm/src/phyltree/PhyTreeGeneratorTask.cpp

```diff
@@ -29,91 +29,70 @@
 
 namespace U2 {
 
-PhyTreeGeneratorTask::PhyTreeGeneratorTask(const MultipleSequenceAlignment& ma, const CreatePhyTreeSettings& _settings)
-    : Task(PhyTreeGeneratorTask::tr("Calculating Phylogenetic Tree"), TaskFlag_FailOnSubtaskError), inputMA(ma), settings(_settings) {
+PhyTreeGeneratorTask::PhyTreeGeneratorTask(const MultipleSequenceAlignment& ma, const CreatePhyTreeSettings& _settings, const TaskFlags& taskFlags)
+    : Task(PhyTreeGeneratorTask::tr("Calculating Phylogenetic Tree"), taskFlags), inputMA(ma), settings(_settings) {
     tpm = Task::Progress_Manual;
 }
 
-void PhyTreeGeneratorTask::run() {
+const PhyTree& PhyTreeGeneratorTask::getResult() const {
+    return result;
 }
 
-Task::ReportResult PhyTreeGeneratorTask::report() {
-    return ReportResult_Finished;
+const CreatePhyTreeSettings& PhyTreeGeneratorTask::getSettings() const {
+    return settings;
 }
 
 PhyTreeGeneratorLauncherTask::PhyTreeGeneratorLauncherTask(const MultipleSequenceAlignment& ma, const CreatePhyTreeSettings& _settings)
-    : Task(PhyTreeGeneratorLauncherTask::tr("Calculating Phylogenetic Tree"), TaskFlag_FailOnSubtaskError), inputMA(ma->getCopy()), settings(_settings), task(nullptr) {
+    : Task(PhyTreeGeneratorLauncherTask::tr("Calculating Phylogenetic Tree"), TaskFlag_NoRun | TaskFlag_FailOnSubtaskError),
+      inputMA(ma->getCopy()), settings(_settings), task(nullptr) {
     tpm = Task::Progress_SubTasksBased;
 }
-void PhyTreeGeneratorLauncherTask::prepare() {
-    QString algId = settings.algorithm;
-    PhyTreeGeneratorRegistry* registry = AppContext::getPhyTreeGeneratorRegistry();
-    PhyTreeGenerator* generator = registry->getGenerator(algId);
-    if (generator == nullptr) {
-        stateInfo.setError(PhyTreeGeneratorLauncherTask::tr("Tree construction algorithm %1 not found").arg(algId));
-    } else {
-        const QStringList& rowsOrder = settings.rowsOrder;
-        if (rowsOrder.size() >= inputMA->getRowNames().size()) {
-            inputMA->sortRowsByList(rowsOrder);
-        }
 
-        namesConvertor.replaceNamesWithAlphabeticIds(inputMA);
+static const QString RENAMED_ROW_PREFIX = "r";
 
-        task = dynamic_cast<PhyTreeGeneratorTask*>(generator->createCalculatePhyTreeTask(inputMA, settings));
-        addSubTask(task);
+void PhyTreeGeneratorLauncherTask::prepare() {
+    PhyTreeGeneratorRegistry* registry = AppContext::getPhyTreeGeneratorRegistry();
+    PhyTreeGenerator* generator = registry->getGenerator(settings.algorithm);
+    CHECK_EXT(generator != nullptr, stateInfo.setError(PhyTreeGeneratorLauncherTask::tr("Tree algorithm %1 is not found").arg(settings.algorithm)), );
+
+    // Assign unique names to rows. The row name is a string representation of the index + 'row_' prefix.
+    originalRowNameByIndex = inputMA->getRowNames();
+    int rowsCount = inputMA->getRowCount();
+    for (int i = 0; i < rowsCount; i++) {
+        QString uniqueRowName = RENAMED_ROW_PREFIX + QString::number(i);
```

Comment:
I suppose, it's better to use `i+1`. The rolled prefix usually starts with 1, not with 0

Reply:
There is nothing about rolling here. This is the index which starts with 0. If I use +1 here I have to use -1 later that has no sense